src/python/zquantum/core/bitstring_distribution.py

```diff
@@ -208,20 +209,22 @@ def get_qubits_number(self):
 
 
 def compute_clipped_negative_log_likelihood(
```

Comment:
I'm wondering if we shouldn't move this to a separate directory?
We need to have MMD in a separate file which is called `distance_measures.py` but then this function is still in `bitstring_distribution.py`.
Perhaps it would make sense to actually put it all into `bistring_distribution` directory and the change into separate files?
WDYT @simonwa7 ?

Reply:
Agreed - I think `bitstring_distribution` should be a directory with:
- `zquantum/core/bitstring_distribution/__init__.py` 
- `zquantum/core/bitstring_distribution/bitstring_distribution.py` (or a similar name to house the `BitstringDistribution` class

Then there should be a `distance_measures` subdirectory with:
- `zquantum/core/bitstring_distribution/distance_measures/clipped_negative_log_likelihood.py`
- `zquantum/core/bitstring_distribution/distance_measures/mmd.py`
- `zquantum/core/bitstring_distribution/distance_measures/utils.py`.

Reply:
So I tried to refactor (see [here](https://github.com/zapatacomputing/z-quantum-core/pull/63/commits/b220e44fd95457990a4d387b396cbe7e1dce6d43))
I don't see why there should be the `zquantum/core/bitstring_distribution/distance_measures/utils.py` file, as I think all the utils are related to the `BitstringDistribution` class and I would leave them there, but happy to hear what you think about this @simonwa7 @mstechly

Reply:
This looks good to me - agreed on placement of `utils`. @mstechly if you're good with this then I think it's good to merge!